lib/autoload.fish

```diff
@@ -10,18 +10,46 @@
 #   Returns != 0 if all paths are missing.
 
 function autoload -d "autoload a function or completion path"
-  for path in $argv
+  set -l paths $argv
+
+  switch "$argv[1]"
+  case '-e' '--erase'
+    set erase
+
+    if test (count $argv) -ge 2
+      # Unshift path to remove option
+      set paths $argv[2..-1]
+    else
+      # Option called without arguments
+      return 1
+    end
+  end
+
+  for path in $paths
+
+    not test -d "$path";
+      and continue
+
+    set path_exist
+
     set -l dest fish_function_path
 
-    if test -d "$path"
-      set path_exist
+    if test (basename "$path") = completions
+      set dest fish_complete_path
+    end
 
-      if test (basename "$path") = completions
-        set dest fish_complete_path
-      end
+    set -l function_path $$dest
 
-      contains "$path" $$dest; or set $dest "$path" $$dest
+    if set -q erase
+      not contains -- "$path" $function_path; and continue
+      set -l index (contains -i -- $path $function_path)
+      set -e function_path[$index]
+    else
+      contains -- "$path" $function_path; and continue
+      set function_path "$path" $function_path
     end
+
+    set $dest $function_path
   end
 
   set -q path_exist
```

Comment:
I don't believe this function return is correct for `autoload -e`. Might want to document it too.